<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.5.0 -->
<title>Sequence-based ab initio prediction of variant effects on expression and disease risk using ExPecto | BayesLabs blog</title>
<meta name="generator" content="Jekyll v3.8.5" />
<meta property="og:title" content="Sequence-based ab initio prediction of variant effects on expression and disease risk using ExPecto" />
<meta name="author" content="Pankaj Katkar" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Understading disease from the DNA mutations is one of the major problem researchers are trying to solve. In this blog, we will analyze about a machine learning model which can predict disease risk from the sequence variation." />
<meta property="og:description" content="Understading disease from the DNA mutations is one of the major problem researchers are trying to solve. In this blog, we will analyze about a machine learning model which can predict disease risk from the sequence variation." />
<link rel="canonical" href="https://bayeslabs.co/2019/07/04/ExPecto.html" />
<meta property="og:url" content="https://bayeslabs.co/2019/07/04/ExPecto.html" />
<meta property="og:site_name" content="BayesLabs blog" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2019-07-04T00:00:00+00:00" />
<script type="application/ld+json">
{"@type":"BlogPosting","url":"https://bayeslabs.co/2019/07/04/ExPecto.html","headline":"Sequence-based ab initio prediction of variant effects on expression and disease risk using ExPecto","dateModified":"2019-07-04T00:00:00+00:00","datePublished":"2019-07-04T00:00:00+00:00","mainEntityOfPage":{"@type":"WebPage","@id":"https://bayeslabs.co/2019/07/04/ExPecto.html"},"author":{"@type":"Person","name":"Pankaj Katkar"},"description":"Understading disease from the DNA mutations is one of the major problem researchers are trying to solve. In this blog, we will analyze about a machine learning model which can predict disease risk from the sequence variation.","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="https://bayeslabs.co/feed.xml" title="BayesLabs blog" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">BayesLabs blog</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/home/">Home</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        
<!-- https://docs.mathjax.org/en/latest/configuration.html#local-config-files -->
<script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/latest.js?config=TeX-AMS-MML_HTMLorMML,https://idcrook.github.io/assets/js/MathJaxLocal.js">
</script>


<article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">Sequence-based ab initio prediction of variant effects on expression and disease risk using ExPecto</h1>
    <p class="post-meta">
      <time class="dt-published" datetime="2019-07-04T00:00:00+00:00" itemprop="datePublished">Jul 4, 2019
      </time>
        â€¢ <span itemprop="author" itemscope itemtype="http://schema.org/Person"><span class="p-author h-card" itemprop="name"><a href="https://www.linkedin.com/in/pankaj-katkar-b0325914a/">Pankaj Katkar</a></span></span></p>
  </header>

  <div class="post-content e-content" itemprop="articleBody">
    <blockquote>
  <p>Understading disease from the DNA mutations is one of the major problem researchers are trying to solve. In this blog, we will analyze about a machine learning model which can predict disease risk from the sequence variation.</p>
</blockquote>

<h2>Introductory note:</h2>
<p>Human DNA sequence consists of genes and regulatory sequences. Genes are used for making proteins while regulatory sequences, also known as non-coding variant act as a regulatory body which decides many genomic functions. The function includes the time utilized in creating the protein, the quantity required for its formation. Today, the biggest challenge in human genetics is whether precision medicine and evolutionary biology are deciphering the regulatory code of gene expression. It also includes understanding the transcriptional effects of genome variation. To cope with this complication, we have used deep learning-based framework known as ExPecto. ExPecto can predict ab initio from the DNA sequence, the tissue-specific transcriptional effects of mutations, including rare or never observed.</p>

<h2>ExPecto:</h2>
<p>ExPecto model takes DNA sequence as input and predicts the future risk of a person from a disease. It uses a deep learning network model followed by spatial transformation and finally, it makes a tissue-specific prediction using L2-regularized linear regression models. The model is fitted by a gradient boosting algorithm. Thus, after getting a tissue-specific prediction, we infer the risk of disease which the patient may have in the future.</p>

<h2>Model planning:</h2>
<figure class="image">
  <img align="center" src="/assets/img/gen3.jpg" alt="Deep learning-based sequence model accurately predicts cell type-specific gene expression" width="" height="" />
  <figcaption>Deep learning-based sequence model accurately predicts cell type-specific gene expression</figcaption>
</figure>

<h2>Input:</h2>
<p>ExPecto takes a DNA sequence of length 40,000. We find the position of allele and then take the DNA sequence of length 20,000 on both left and right side. If we take less than 40k size, the performance decreases slightly. But if we increase the size and it is more than 40k, there is an insignificant chance of high performance. You will witness that in the below figure.</p>
<figure class="image">
  <img align="center" src="/assets/img/gen.png" alt="Comparison of ExPecto eQTL prediction performance across models trained with different sequence window sizes." width="" height="" />
  <figcaption>Comparison of ExPecto eQTL prediction performance across models trained with different sequence window sizes.</figcaption>
</figure>

<p>In the above figure, each line shows performance for the model trained with different window sizes (+/- around the TSS).  The x-axis represents the predicted effect magnitude cutoff, as measured by absolute log fold-change. Y-axis represents the accuracy of predicting the expression change directionality for the variants above the corresponding effect magnitude.</p>

<h2>Architecture:</h2>
<p>ExPecto model consist of three components that act sequentially.<br /></p>

<h4>1) The deep convolutional network regulatory feature representation model architecture:</h4>
<p>In total, we have 2002 genome-wide histone marks, transcription factor binding and chromatin accessibility features. Our deep learning model scans the sequences each of the length 2000. The deep convolutional neural network model predicts epigenomic features of a 200bp region. It also uses the 1800bp surrounding context sequence. Thus each 200 bp region will give 2002 features, thus ranging in a total of  400400 features.
Following is the architecture of deep convolutional network:<br />
Input (Size: 4bases x 2000bp) =&gt;<br />
    (Layer 1): Convolution(4 -&gt; 320, kernel size=8)<br />
    (Layer 2): ReLU<br />
    (Layer 3): Convolution(320 -&gt; 320, kernel size=8)<br />
    (Layer 4): ReLU<br />
    (Layer 5): Dropout(Probability=0.2)<br />
    (Layer 6): Max pooling(pooling size=4)<br />
    (Layer 7): Convolution(320 -&gt; 480, kernel size=8)<br />
    (Layer 8): ReLU<br />
    (Layer 9): Convolution(480 -&gt; 480, kernel size=8)<br />
    (Layer 10): ReLU<br />
    (Layer 11): Dropout(Probability=0.2)<br />
    (Layer 12): Max pooling(pooling size=4)<br />
    (Layer 13): Convolution(480 -&gt; 960, kernel size=8)<br />
    (Layer 14): ReLU<br />
    (Layer 15): Convolution(960 -&gt; 960, kernel size=8)<br />
    (Layer 16): ReLU<br />
    (Layer 17): Dropout(Probability=0.2)<br />
    (Layer 18): Linear(101760 -&gt; 2003)<br />
    (Layer 19): ReLU<br />
    (Layer 20): Linear(2003 -&gt; 2002)<br />
    (Layer 21): Sigmoid<br />
    =&gt; Output (Size: 2002 epigenomic features)<br /></p>

<p>ReLU indicates the rectified linear unit activation function. Sigmoid indicates the Sigmoid activation function.  Notations such as 4 -&gt; 320 indicate the input and output channel size for each layer.</p>

<h4> 2) Spatial Feature Transformation: </h4>
<p>The second component of ExPecto is the spatial transformation module that reduces the dimensionality of the learning problem by generating spatially-transformed features. The spatial transformation module reduces the input dimensionality with ten exponential functions weighting upstream and downstream regions separately. The weights are based on the relative distance to TSS (transformed features with a higher decay rate are more concentrated near TSSs). This effectively reduces the number of features from 20 fold to 20020. The exponential functions were prespecified (based on empirical selection) and not learned during training.
Scanning the 40kb region surrounding the TSS with the deep convolutional network model outputs 200 spatial bins for each of the 2002 histone mark, transcription factor, and DNase features. Spatial transformation condenses this information into 10 spatial features by using exponential functions with 5 different decay rates to compute the weighted sum over spatial bins. Upstream regions and downstream regions are transformed independently.</p>
<figure class="image">
  <img align="center" src="/assets/img/gen1.png" alt="Schematic representation of spatial transformation in ExPecto. " width="" height="" />
  <figcaption>Schematic representation of spatial transformation in ExPecto. </figcaption>
</figure>

<h4> 3) L2-regularized linear regression model: </h4>
<p>Finally, to make tissue-specific expression predictions, spatially-transformed features are used to predict gene expression levels. For each tissue, L2-regularized linear regression models fitted by gradient boosting algorithm are used. Specifically, the full models including both spatial transformation and linear models are specified as below.</p>

<script type="math/tex; mode=display">% <![CDATA[
expression = \sum_{d \in D}\sum_{i} p_{id}  \Bigg[ \sum_{k} 1(t_{d} < 0) \beta^{up}_{ik} e^{-a_{k} \left\lfloor\dfrac{\abs{t_{d}}}{200bp}\right\rfloor} + \sum_{k} 1(t_{d} > 0) \beta^{down}_{ik} e^{-a_{k} \left\lfloor\dfrac{\abs{t_{d}}}{200bp}\right\rfloor} \Bigg] %]]></script>

<h2>Summary:</h2>
<p>So using this model we can predict the risk of disease in a person so that he can start with his treatment immediately.</p>

  </div><div id="disqus_thread"></div>
<script>

var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};

(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://bayeslabs.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a class="u-url" href="/2019/07/04/ExPecto.html" hidden></a>
</article>
<div id="disqus_thread"></div>
<script>


      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">BayesLabs blog</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">BayesLabs blog</li><li><a class="u-email" href="mailto:contact@bayeslabs.co">contact@bayeslabs.co</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/bayeslabs"><svg class="svg-icon"><use xlink:href="/assets/minima-social-icons.svg#github"></use></svg> <span class="username">bayeslabs</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p></p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
